### ISSUE TITLE
Issue with opening Chinese document properly using pdf.js

### ISSUE THREAD

**Issue by sandeepsuthari** [mozilla/pdf.js#12327]
Attach (recommended) or Link to PDF file here:
[China EE_Handbook_China_JustSomeIndexContent.pdf](https://github.com/mozilla/pdf.js/files/5175484/China.EE_Handbook_China_JustSomeIndexContent.pdf)

Configuration:
- Web browser and its version:
- Operating system and its version:
- PDF.js version:
- Is a browser extension:

Steps to reproduce the problem:
1. Try embedding the pdf.js and start using it for PDF viewing in the application.And try viewing the attached Chinese document, you will see its not getting [opened](url)
2. You will see the following issues while inspecting in Google chrome using the pdf.js chrome extension :

**_Access to XMLHttpRequest at 'file:///Users/sandeep.suthari/SandeepSuthari/Softwares/pdfjs-2.5.207-es5-dist/web/cmaps/Adobe-GB1-UCS2.bcmap' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
util.js:336 Warning: Error during font loading: Unable to load binary CMap at: ../web/cmaps/Adobe-GB1-UCS2.bcmap_**

What is the expected behavior? (add screenshot)
The PDF attachment should be rendered properly

What went wrong? (add screenshot)
<img width="1680" alt="Screenshot 2020-09-04 at 8 12 44 PM" src="https://user-images.githubusercontent.com/8459122/92251827-095c7180-eeeb-11ea-8b4a-38a5d2fe25de.png">

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

**Comment by Snuffleupagus** [comment]
> Access to XMLHttpRequest at 'file:///Users/sandeep.suthari/SandeepSuthari/Softwares/pdfjs-2.5.207-es5-dist/web/cmaps/Adobe-GB1-UCS2.bcmap' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
> util.js:336 Warning: Error during font loading: Unable to load binary CMap at: ../web/cmaps/Adobe-GB1-UCS2.bcmap

The error message clearly points to the *browser* being responsible for the blocking, and not the PDF.js library, hence this isn't an actionable/valid issue here; see also https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#faq-xhr

Furthermore, as mentioned in the README, many browsers enforce special restrictions for `file://` URLs which is why you should be using a server; see https://github.com/mozilla/pdf.js/#getting-the-code

**Comment by sandeepsuthari** [comment]
Hi Snuffleupagus, the error details are from the google chrome inspect option, with the embedded Pdf.js also in our product
we are facing the same issue where the PDF is not being rendered

**Comment by sandeepsuthari** [comment]
In our application, following are the error details :
**_PDF e74d03f695f8434bb9dc85375982f3c3 [1.5 - / -] (PDF.js: 1.5.188)
Warning: Error during font loading: Cannot read property 'push' of undefined_.  (pdf.js:230) **

**Comment by Snuffleupagus** [comment]
> with the embedded Pdf.js also in our product
> we are facing the same issue where the PDF is not being rendered

That doesn't change the fact that this is, as already mentioned, a *browser* limitation; hence you should actually look into the CORS issues (however that's not something that anyone here can help you fix).

> PDF e74d03f695f8434bb9dc85375982f3c3 [1.5 - / -] (PDF.js: 1.5.188)_

That version is several *years* out-of-date, and is consequently not supported; please find the current releases at https://github.com/mozilla/pdf.js/releases

> Warning: Error during font loading: Cannot read property 'push' of undefined_. (pdf.js:230) 

That seems totally unrelated to the rest of the issue, and anyway you're using an unsupported version.

---

Furthermore, when opening an issue you should always provide *all* of the information requested in https://github.com/mozilla/pdf.js/blob/master/.github/ISSUE_TEMPLATE.md; as-is this issue contains no actionable information.

Also, note https://github.com/mozilla/pdf.js/blob/master/.github/CONTRIBUTING.md (emphasis mine):
> If you are developing a custom solution, first check the examples at https://github.com/mozilla/pdf.js#learning and search existing issues. If this does not help, **please prepare a short well-documented example that demonstrates the problem and make it accessible online** on your website, JS Bin, GitHub, etc. before opening a new issue or contacting us in the Matrix room -- **keep in mind that just code snippets won't help us troubleshoot the problem**.

Finally, please see http://mozilla.github.io/pdf.js/getting_started/#introduction (emphasis mine):
> The viewer is built on the display layer and is the UI for PDF viewer in Firefox and the other browser extensions within the project. It can be a good starting point for building your own viewer. ***However, we do ask if you plan to embed the viewer in your own site, that it not just be an unmodified version. Please re-skin it or build upon it.***
